### PR TITLE
added robin actions

### PR DESCRIPTION
**Pull request by proksh**

### CODE REVIEW

File: ApiClient.ts [hunk 1]
```diff
@@ -1,134 +1,175 @@
...
+        },
+        body: formData,
+      });
+      if (!res.ok) {
+        const body = await res.text();
+        throw new Error(
+          `Request to ${res.url} failed (${res.status}): ${body}`
+        );
+      }
+      return (await res.json()) as UploadResponse;
```
Comment (amanjeetsingh150) [inline ApiClient.ts:105]: This should be according to RobinUploadResponse

**Review by amanjeetsingh150 — APPROVED** [review]

File: ApiClient.ts [hunk 2]
```diff
@@ -1,134 +1,203 @@
...
+  iOSVersion?: number;
+  includeTags: string[];
+  excludeTags: string[];
+  appBinaryId?: string;
+  deviceLocale?: string;
+};
+
+type MaestroUploadRequest = Omit<
+  CloudUploadRequest,
+  "benchmarkName" | "repoOwner" | "repoName" | "agent"
```
Comment (amanjeetsingh150) [inline ApiClient.ts:32]: We do have repoOwner, repoName and agent on Robin Request. Also I think maybe comment which request is for robin or new infra

**Review by bartekpacia — COMMENTED** [review]
I'd prefer to see all `"` -> `'` changes in a separate PR. It makes it hard to review.

File: ApiClient.ts (hunk 2 above)
Comment (proksh) [reply ApiClient.ts]: I tried - "repoOwner" is giving an error.

Added back "repoName" and "agent".

**Comment by proksh** [comment]
> I'd prefer to see all `"` -> `'` changes in a separate PR. It makes it hard to review.

Done

File: ApiClient.ts (hunk 2 above)
Comment (amanjeetsingh150) [reply ApiClient.ts]: Strange 🤔 What error?

Comment (proksh) [reply ApiClient.ts]: Fixed with the changes on copilot. 🙌

**Pull request merged**